File: src/test/java/dev/retrotv/crypto/owe/kdf/bcrypt/BCryptTest.java
package dev.retrotv.crypto.owe.kdf.bcrypt;

import dev.retrotv.crypto.owe.OWETest;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class BCryptTest extends OWETest {

    @Test
    @DisplayName("BCrypt password encode 테스트")
    void passwordEncrypt() {
        passwordEncryptAndMatchesTest(new BCrypt());
    }

    @Test
    @DisplayName("upgradeEncoding 테스트")
    void test_upgradeEncoding_method() {
        BCrypt bCrypt = new BCrypt();
        assertTrue(bCrypt.upgradeEncoding("!Q@W3e4r"));
    }
}

File: src/test/kotlin/dev/retrotv/crypto/owe/kdf/bcrypt/BCryptTest.kt
package dev.retrotv.crypto.owe.kdf.bcrypt

import dev.retrotv.crypto.owe.OWETest
import dev.retrotv.crypto.owe.password.bcrypt.BCrypt
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder.BCryptVersion
import java.security.SecureRandom

class BCryptTest : OWETest() {

    @Test
    @DisplayName("BCrypt password encode 테스트")
    fun passwordEncrypt() {
        passwordEncryptAndMatchesTest(BCrypt())
        passwordEncryptAndMatchesTest(BCrypt(10))
        passwordEncryptAndMatchesTest(BCrypt(BCryptVersion.`$2A`))
        passwordEncryptAndMatchesTest(BCrypt(BCryptVersion.`$2A`, SecureRandom()))
        passwordEncryptAndMatchesTest(BCrypt(10, SecureRandom()))
        passwordEncryptAndMatchesTest(BCrypt(BCryptVersion.`$2A`, 10))
        passwordEncryptAndMatchesTest(BCrypt(BCryptVersion.`$2A`, 10, SecureRandom()))
    }
}
